### PR TITLE
GCW-2704 SMS fallback SHOULD be sent if the offer is cancelled or closed

### PR DESCRIPTION
**Pull request by abhinav098**
Hello team, 
This PR adds rejected items to the list of the sms_reminders sending list, so that the users who are not on our app and item gets rejected, they should receive the fallback message regarding the same. 

Please review.
Thanks

### CODE REVIEW

**Comment by patrixr** [comment]
@abhinav098 Could you hold off on merging this. Had a chat with @steveyken and I think we need to  review the intended behaviour a bit. We'll clear up things over scrum tomorrow :)

**Comment by abhinav098** [comment]
Sure, @steveyken , sorry for misunderstanding. Will make the necessary changes and let you know.

**Comment by abhinav098** [comment]
@steveyken , I have made the changes please review this PR now, I have made the necessary changes.

File: app/services/subscriptions_reminder.rb [hunk 1]
```diff
@@ -22,17 +22,17 @@ def generate
   # IMPORANT NOTE: When a user submits an offer, we set the sms_reminder_sent_at time to now + 1.minute
   #   to avoid alerting on the system 'thank you for submitting your offer' message
   def user_candidates_for_reminder
-    offer_states = ['submitted', 'under_review', 'reviewed', 'scheduled', 'received', 'receiving', 'inactive'] # NOT draft, closed or cancelled
+    offer_states = ['submitted', 'under_review', 'reviewed', 'scheduled', 'received', 'receiving', 'inactive', 'closed', 'cancelled']
```
Comment (namrataukirde) [inline app/services/subscriptions_reminder.rb]: Keep this comment => # NOT draft

File: app/services/subscriptions_reminder.rb [hunk 2]
```diff
@@ -22,17 +22,17 @@ def generate
...
-        .where("COALESCE(users.sms_reminder_sent_at, users.created_at) < (?)", delta.iso8601)
-        .where('subscriptions.state': 'unread')
-        .where("messages.created_at > COALESCE(users.sms_reminder_sent_at, users.created_at)")
-        .where("(messages.offer_id IS NOT NULL OR messages.item_id IS NOT NULL) and messages.order_id IS NULL")
-        .where("offers.created_by_id = subscriptions.user_id")
-        .where("offers.state IN (?)", offer_states)
-        .where('messages.sender_id != offers.created_by_id')
-        .where("messages.created_at < (?)", head_start.iso8601)
-        .distinct
+      .where("COALESCE(users.sms_reminder_sent_at, users.created_at) < (?)", delta.iso8601)
```
Comment (namrataukirde) [inline app/services/subscriptions_reminder.rb]: Check indentation issues of code climate. :)

File: app/services/subscriptions_reminder.rb (hunk 1 above)
Comment (steveyken) [reply app/services/subscriptions_reminder.rb]: yes, include not draft!

**Comment by namrataukirde** [comment]
Sure @steveyken. we will do demo run for same on staging tomorrow. By having slack notifications thing. We will merge PR tomorrow and confirm it on staging first. 👍

**Pull request merged**